storage/pebble/my_receipts.go

```diff
@@ -1,74 +1,56 @@
-package badger
+package pebble
 
 import (
-	"errors"
 	"fmt"
 
-	"github.com/dgraph-io/badger/v2"
+	"github.com/cockroachdb/pebble"
 
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/module"
 	"github.com/onflow/flow-go/module/metrics"
 	"github.com/onflow/flow-go/storage"
-	"github.com/onflow/flow-go/storage/badger/operation"
-	"github.com/onflow/flow-go/storage/badger/transaction"
+	"github.com/onflow/flow-go/storage/pebble/operation"
 )
 
 // MyExecutionReceipts holds and indexes Execution Receipts.
-// MyExecutionReceipts is implemented as a wrapper around badger.ExecutionReceipts
+// MyExecutionReceipts is implemented as a wrapper around pebble.ExecutionReceipts
 // The wrapper adds the ability to "MY execution receipt", from the viewpoint
 // of an individual Execution Node.
 type MyExecutionReceipts struct {
 	genericReceipts *ExecutionReceipts
-	db              *badger.DB
+	db              *pebble.DB
 	cache           *Cache[flow.Identifier, *flow.ExecutionReceipt]
 }
 
-// NewMyExecutionReceipts creates instance of MyExecutionReceipts which is a wrapper wrapper around badger.ExecutionReceipts
+// NewMyExecutionReceipts creates instance of MyExecutionReceipts which is a wrapper wrapper around pebble.ExecutionReceipts
 // It's useful for execution nodes to keep track of produced execution receipts.
-func NewMyExecutionReceipts(collector module.CacheMetrics, db *badger.DB, receipts *ExecutionReceipts) *MyExecutionReceipts {
-	store := func(key flow.Identifier, receipt *flow.ExecutionReceipt) func(*transaction.Tx) error {
+func NewMyExecutionReceipts(collector module.CacheMetrics, db *pebble.DB, receipts *ExecutionReceipts) *MyExecutionReceipts {
+	store := func(key flow.Identifier, receipt *flow.ExecutionReceipt) func(storage.PebbleReaderBatchWriter) error {
 		// assemble DB operations to store receipt (no execution)
 		storeReceiptOps := receipts.storeTx(receipt)
 		// assemble DB operations to index receipt as one of my own (no execution)
 		blockID := receipt.ExecutionResult.BlockID
 		receiptID := receipt.ID()
-		indexOwnReceiptOps := transaction.WithTx(func(tx *badger.Txn) error {
-			err := operation.IndexOwnExecutionReceipt(blockID, receiptID)(tx)
-			// check if we are storing same receipt
-			if errors.Is(err, storage.ErrAlreadyExists) {
```

Comment:
## ⚠️ 

Are you sure that is a wise idea. You are the expert, but if I remember correctly we previously had execution nodes crash telling us that they were attempting to override their own result with a different one. With this check removed, they wouldn't notice that anymore. 

I think we have options here: 
* Wouldn't including a mutex suffice to prevent dirty reads? Then we can check first, because no other components would be able to write (ideally, we would move all the write operations related to ExecutionReceipts into this package not not export them.